internal/clients/config/kibana.go

```diff
@@ -0,0 +1,101 @@
+package config
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	"github.com/disaster37/go-kibana-rest/v8"
+	fwdiags "github.com/hashicorp/terraform-plugin-framework/diag"
+	sdkdiags "github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type kibanaConfig kibana.Config
+
+func newKibanaConfigFromSDK(d *schema.ResourceData, base baseConfig) (kibanaConfig, sdkdiags.Diagnostics) {
+	var diags sdkdiags.Diagnostics
+
+	kibConn, ok := d.GetOk("kibana")
+	if !ok {
+		return kibanaConfig{}, diags
+	}
+
+	// Use ES details by default
+	config := base.toKibanaConfig()
```

Comment:
It looks combersome. The logic allows creating Kibana resources using creds from ES config and endpoints definition from an env variable but do we really need this?
It sounds like a legacy but maybe we can refactor that - IMO it complicates the logic without real benefits. Kibana config should be taken from `kibana` atrribute and the corresponding env vars. 
The similar approach is used for reading Fleet config.

Reply:
> IMO it complicates the logic without real benefits.

Whether there's benefit would depend heavily on the usage, if people are using the provider to configure both Kibana and ES resources then sharing credentials simplifies their configuration. FWIW I have only ever configured the ES credentials and relied on Kibana using those credentials.